server/tests/test_notebook_view.py

```diff
@@ -149,6 +149,25 @@ def test_tryit_view(client, fake_user, logged_in, iomd):
         assert len(response.redirect_chain) == 0
 
 
+@pytest.mark.parametrize("logged_in", [True, False])
+def test_tryit_view_get_api(client, fake_user, logged_in):
+    """
+    Test if /try-it handles GET request file upload correctly
+    """
+
+    import mimetypes
+
+    file_name = "data.txt"
+    file_content = "testing"
+    file_mimetype = mimetypes.guess_type(file_name)
+
+    if logged_in:
+        client.force_login(fake_user)
+
+    resp = client.get(f"/tryit?file={file_content}&filename={file_name}", follow=True)
+    assert get_file_script_block(resp.content, file_name, file_mimetype) == file_content
```

Comment:
Looking again, this is only correct if `logged_in` is false. If it is true, you check that the file is created as expected:

https://github.com/iodide-project/iodide/blob/54d127d7f7ff1d2e4bb2c3049e33c75805e745c9/server/notebooks/views.py#L181

In this case there should be no file script block (since the files are already created on the server, there is no need to provide "fake" files in script blocks to the client)